netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -368,8 +368,10 @@ _net_ebpf_extension_hook_provider_attach_client(
         *provider_binding_context = hook_client;
         hook_client = NULL;
     } else {
-        if (hook_client)
```

Comment:
I would rather have a function to clean up hook_client including fields such as work items.

Reply:
fixed.